app/src/main/java/com/runnect/runnect/util/custom/RequireLoginDialogFragment.kt

```diff
@@ -0,0 +1,40 @@
+package com.runnect.runnect.util.custom
+
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import com.runnect.runnect.R
+import com.runnect.runnect.binding.BindingDialogFragment
+import com.runnect.runnect.databinding.FragmentRequireLoginDialogBinding
+import com.runnect.runnect.presentation.login.LoginActivity
+
+class RequireLoginDialogFragment
+    : BindingDialogFragment<FragmentRequireLoginDialogBinding>(R.layout.fragment_require_login_dialog) {
```

Comment:
저는 Dialog를 확장함수로 만들어서 썼는데 이렇게 class를 만들어 쓰는 것과 비교해서 어떤 것이 더 낫다고 생각하시는지 의견 궁금합니다! (어떤 특정 방법을 강요하는 것이 아닌 discussion의 목적으로 물어보는 것)

제가 만든 Dialog 확장함수는 setActivityDialog, setFragmentDialog입니다!

Reply:
AlertDialog vs. DialogFragment 차이점에 대한 질문으로도 생각해볼 수 있는데요! 

https://stackoverflow.com/questions/13765127/dialogfragment-advantages-over-alertdialog

위의 글을 읽어보니 DialogFragment는 프래그먼트이다보니 생명주기에 따라 처리를 해줄 수 있다는 게 큰 장점인 거 같아요! 

>Have you ever seen window leaks if you didn't close a dialog when its Activity was getting destroyed? 
So to prevent that, have you ever tried to close the dialog when onPause() was called? 
So to do that, have you ever had to make a reference of that dialog to a class level object?
With DialogFragment, it's all handled. And you get all lifecycle callbacks. Then you can provide more intelligence to the dialog and make it do some smart work on its own rather than Activity telling it what to do.

이거 읽고 저번 회의 때 언급했던 WindowLeak 에러가 났던 이유를 알 수 있었습니다,, ㅎㅎ

프래그먼트 생명주기에 따라 처리를 해줄 수 있는 DialogFragment를 사용하는 게 더 좋을 거 같습니다!

Reply:
Activity가 소멸됐는데도 dialog는 남아있어서 에러가 떴던 거군요. 감사합니다!

Reply:
Activity 하위에 DialogFragment를 만들면 해당 Activity가 소멸될 때 DialogFragment도 자동으로 객체 소멸이 되나요?

Reply:
AlertDialog가 아닌 DialogFragment를 썼을 때의 단점은 뭐가 있을까요? (discussion 목적으로 물어보는 것)

Reply:
>액티비티와 프래그먼트는 각각 독립적인 생명주기를 가지고 있지만, 프래그먼트는 액티비티에 종속적인 존재입니다. 
따라서 액티비티가 파괴되면, 그에 연결된 프래그먼트들도 같이 파괴되는 것이 기본적인 동작입니다.
하지만 이러한 동작은 프래그먼트를 사용하는 방식이나 상황에 따라 다르게 처리될 수 있습니다. 예를 들어, 프래그먼트를 백스택에 추가하거나, setRetainInstance(true)를 호출하여 인스턴스 상태를 유지하도록 설정하면, 액티비티가 파괴되어도 프래그먼트는 파괴되지 않을 수 있습니다. 이런 경우, 액티비티가 다시 생성될 때 원래의 프래그먼트 인스턴스를 재사용할 수 있습니다.

gpt 설명이 맞는 거 같습니다! ㅎㅎ

Reply:
> 안드로이드에서 AlertDialog와 DialogFragment는 화면에 다이얼로그를 표시하는 데 사용되는 두 가지 주요 방법입니다. 이들은 각각 다음과 같은 장단점을 가지고 있습니다. 
>
>**AlertDialog**
> - 장점: AlertDialog는 간단하고 빠르게 구현할 수 있는 장점이 있습니다. 코드를 적게 사용하므로 빠르게 다이얼로그를 생성하고 표시할 수 있습니다.
> - 단점: 하지만 생명주기 관리 측면에서는 단점이 있습니다. AlertDialog는 그 자체로 생명주기를 가지지 않아서, 화면 회전이나 액티비티의 종료 등의 상황에서 다이얼로그가 자동으로 관리되지 않습니다. 따라서 액티비티가 파괴될 때 다이얼로그를 닫지 않으면 윈도우 누수(window leak)가 발생할 수 있습니다. 이를 방지하기 위해 onPause()가 호출될 때 다이얼로그를 닫는 등의 추가 작업이 필요합니다. 이를 위해선 다이얼로그에 대한 참조를 클래스 레벨 객체로 유지해야 할 수도 있습니다.
>
>**DialogFragment**
> - 장점: DialogFragment는 프래그먼트의 생명주기를 따르므로, 화면 회전이나 액티비티의 종료 등의 상황에서 자동으로 관리됩니다. 따라서 윈도우 누수와 같은 문제를 자동으로 방지할 수 있습니다. 또한 모든 생명주기 콜백을 받게 되므로, 다이얼로그에 더 많은 지능을 제공하고 액티비티 대신 스스로 일을 처리하도록 만들 수 있습니다.
> - 단점: 그러나 DialogFragment의 구현은 AlertDialog보다 복잡할 수 있습니다. 프래그먼트의 생명주기를 이해하고 관리해야 하므로 코드가 복잡해질 수 있습니다.
>
> 따라서 어떤 방식을 선택할지는 사용 상황과 개발자의 선호에 따라 달라질 수 있습니다. 간단한 다이얼로그를 빠르게 표시하려면 AlertDialog를, 생명주기 관리와 윈도우 누수 방지 등의 장점을 원한다면 DialogFragment를 사용하는 것이 좋을 수 있습니다.

gpt가 설명을 참 잘하네요..^^ 제가 하고 싶은 말 그대로 정리해놨어요!

Reply:
굿